test/integration/test-custom-filename.js

```diff
@@ -20,25 +20,26 @@ var server = http.createServer(function(req, res) {
 
   var form = new IncomingForm({uploadDir: common.dir.tmp});
 
-  form.parse(req);
-
-  form
-    .on('file', function(name, file) {
-      assert.strictEqual(name, 'my_file');
-      assert.strictEqual(file.name, options.filename);
-      assert.strictEqual(file.type, options.contentType);
-    })
-    .on('end', function() {
+  form.parse(req, function (err, fields, files) {
+      assert('my_file1' in files);
+      assert.strictEqual(files['my_file1'].name, options.filename);
+      assert.strictEqual(files['my_file1'].type, options.contentType);
+
+      assert('my_file2' in files);
+      assert.strictEqual(files['my_file2'].name, options.filename);
+      assert.strictEqual(files['my_file2'].type, 'image/png');
```

Comment:
Because `options.filename` is `test.png` and is being parsed by mimetype.

In my_file1 it is explicitly set to `image/gif`, however my_file2 gets it's content-type from the file extension.